shared-libraries.rst

```diff
@@ -0,0 +1,23 @@
+Using shared libraries
+======================
+
+**Shared libraries** are libraries that are loosely coupled to the programs and extensions that depend on them. When loading an executable into memory, an operating system will find all dependent shared libraries and link them to the executable so that it can run. The major operating systems Linux, OS X, and Windows all provide a way to build executables and libraries that contain links to the shared libraries they depend on, instead of the directly linking the libraries themselves.
+
+Dynamic links are discovered in a similar manner to the way that python modules are discovered via ``PYTHONPATH``, and executables are discovered via ``PATH``.  A list of search locations is made, then the library objects are searched for in the search locations.  By default, as well as by design, the system dynamic linker does not have any special preference for the conda environment lib directories.
+
+Both absolute links and relative links may be specified. If the links were absolute paths like ``/Users/jsmith/my_build_env`` then the library would only work on a system where that exact path exists. Therefore, relative links are preferred in conda packages.
+
+To use relative links, we need to use a special variable in the link itself.  In Linux, the $ORIGIN variable allows you to say "relative to this file as it's being executed".  On OS X, there are ``@rpath``, ``@loader_path``, ``@executable_path``.  ``@rpath`` allows you to set relative links from the system load paths, while ``@loader_path`` is equivalent to ``$ORIGIN``, and ``@executable_path`` supports the Apple ".app" directory approach, where libraries know where they live relative to their calling application.  In Windows, _____________.
+
+Windows instructions are missing from here onward...............
```

Comment:
Please see comment above for how Windows handles this (TLDR: it is PATH, plus a couple of other locations.  No nice linking like Linux/OS X).  Set PATH dynamically for your application, so that the RIGHT DLL gets found.  Windows is weak in this regard, and we must manage PATH very carefully as a result.